webapp/utils/post_utils.jsx

```diff
@@ -107,3 +107,20 @@ export function shouldShowDotMenu(post) {
 
     return false;
 }
+
+export function contains(text, key) {
+    if (!text || !key) {
+        return false;
+    }
+
+    const pattern = /`${key}[^\w-]`/;
```

Comment:
You'll also want to use the `i` flag to tell this to be case insensitive since `@ALL` also mentions everyone. You can do this with `/something/i` or `new RegExp('something', 'i')` depending on how you're constructing the pattern